{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "gse_list = ['GSE70436', 'GSE76507', 'GSE74278']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "import xml.etree.ElementTree as ET\n",
    "import urllib.request"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [
    {
     "ename": "KeyError",
     "evalue": "'outfile_root'",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mKeyError\u001b[0m                                  Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-26-c32da4816bc0>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m()\u001b[0m\n\u001b[1;32m      1\u001b[0m \u001b[0;32mfor\u001b[0m \u001b[0mgse\u001b[0m \u001b[0;32min\u001b[0m \u001b[0mgse_list\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m----> 2\u001b[0;31m     \u001b[0moutfile\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;34m'{outfile_root}{gse}-annotation.txt'\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mformat\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m**\u001b[0m\u001b[0mlocals\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m      3\u001b[0m     \u001b[0;32mif\u001b[0m \u001b[0;32mnot\u001b[0m \u001b[0mos\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mpath\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mexists\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0moutfile\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      4\u001b[0m         \u001b[0mgeoId\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mET\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mfromstring\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0murllib\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mrequest\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0murlopen\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m'https://eutils.ncbi.nlm.nih.gov/entrez/eutils/esearch.fcgi?db=gds&term={}%5BAccession%20ID%5D'\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mformat\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mgse\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mread\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mfindall\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m'IdList'\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;36m0\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;36m0\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mtext\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      5\u001b[0m         \u001b[0mroot\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mET\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mfromstring\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0murllib\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mrequest\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0murlopen\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m'https://eutils.ncbi.nlm.nih.gov/entrez/eutils/esummary.fcgi?db=gds&id={geoId}'\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mformat\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m**\u001b[0m\u001b[0mlocals\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mread\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mKeyError\u001b[0m: 'outfile_root'"
     ]
    }
   ],
   "source": [
    "for gse in gse_list:\n",
    "    outfile = '{outfile_root}{gse}-annotation.txt'.format(**locals())\n",
    "    if not os.path.exists(outfile):\n",
    "        geoId = ET.fromstring(urllib.request.urlopen('https://eutils.ncbi.nlm.nih.gov/entrez/eutils/esearch.fcgi?db=gds&term={}%5BAccession%20ID%5D'.format(gse)).read()).findall('IdList')[0][0].text\n",
    "        root = ET.fromstring(urllib.request.urlopen('https://eutils.ncbi.nlm.nih.gov/entrez/eutils/esummary.fcgi?db=gds&id={geoId}'.format(**locals())).read())\n",
    "        annotated_dataset = {x.attrib['Name']: x.text for x in root.find('DocSum') if 'Name' in x.attrib.keys() and x.attrib['Name'] in ['title', 'summary']}\n",
    "        with open(outfile, 'w') as openfile:\n",
    "            openfile.write(json.dumps(annotated_dataset))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [],
   "source": [
    "gse = 'GSE70436'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [],
   "source": [
    "geoId = ET.fromstring(urllib.request.urlopen('https://eutils.ncbi.nlm.nih.gov/entrez/eutils/esearch.fcgi?db=gds&term={}%5BAccession%20ID%5D'.format(gse)).read()).findall('IdList')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [],
   "source": [
    "root = ET.fromstring(urllib.request.urlopen('https://eutils.ncbi.nlm.nih.gov/entrez/eutils/esummary.fcgi?db=gds&id={geoId}'.format(**locals())).read())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [],
   "source": [
    "attributes = ['title', 'summary']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [],
   "source": [
    "annotated_dataset = {x.attrib['Name']: x.text for x in root.find('DocSum') if 'Name' in x.attrib.keys() and x.attrib['Name'] in ['title', 'summary']}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'summary': 'The tetrapod-restricted KRAB-containing zinc finger proteins (KRAB-ZFPs) are essential early embryonic controllers of transposable elements (TEs), which they repress via their cofactor KAP1 and associated effectors through histone and DNA methylation, a process thought to result in irreversible silencing. Using a target-centered functional screen, we matched several murine TEs with their cognate KRAB-ZFP. This revealed an unexpected level of granularity in their interactions, with KRAB-ZFPs recognizing TEs from more than one subfamily, TEs recruiting more than one KRAB-ZFP, and spatially and temporally differential KRAB-ZFP-mediated regulation of TEs and nearby genes. Most importantly, we discovered that the KRAB/KAP1 system controls TEs in adult tissues, in cell culture and in vivo, where they partner up to regulate the expression of cellular genes. Therefore, TEs and KRAB-ZFPs establish widely active transcription networks that regulate not only development but probably also many physiological events. Given the high degree of species-specificity of both TEs and KRAB-ZFPs, these results have important implications for studying and understanding the biology of higher vertebrates, including humans.',\n",
       " 'title': 'Transposable elements and their KRAB-ZFP controllers regulate gene expression in adult tissues'}"
      ]
     },
     "execution_count": 25,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "annotated_dataset"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Check if GEO\n",
    "if dataset_accession[:3] in ['GDS', 'GSE']:\n",
    "\n",
    "    url = 'http://eutils.ncbi.nlm.nih.gov/entrez/eutils/esearch.fcgi?db=gds&term={dataset_accession}%5BAccession%20ID%5D'.format(**locals())\n",
    "    print url\n",
    "\n",
    "    print 'opening...'\n",
    "\n",
    "    # Get GEO ID\n",
    "    geoId = ET.fromstring(urllib2.urlopen('https://eutils.ncbi.nlm.nih.gov/entrez/eutils/esearch.fcgi?db=gds&term={dataset_accession}%5BAccession%20ID%5D'.format(**locals())).read()).findall('IdList')[0][0].text\n",
    "\n",
    "    # Get GEO Annotation\n",
    "    root = ET.fromstring(urllib2.urlopen('https://eutils.ncbi.nlm.nih.gov/entrez/eutils/esummary.fcgi?db=gds&id={geoId}'.format(**locals())).read())\n",
    "\n",
    "    # Convert to dictionary\n",
    "    annotated_dataset = {x.attrib['Name'].replace('title', 'dataset_title').replace('summary', 'dataset_description'): x.text.encode('ascii', 'ignore').replace('%', '%%').replace('\"', \"'\") for x in root.find('DocSum') if 'Name' in x.attrib.keys() and x.attrib['Name'] in attributes}\n",
    "\n",
    "    # Get landing URL\n",
    "    annotated_dataset['dataset_landing_url'] = 'https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc='+dataset_accession if dataset_accession[:3] == 'GDS' else 'https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc='+dataset_accession\n",
    "\n",
    "    # Add repository FK\n",
    "    annotated_dataset['repository_fk'] = 1\n",
    "\n",
    "else:\n",
    "\n",
    "    # Return empty\n",
    "    annotated_dataset = {'title': '', 'summary': '', 'repository_name': '', 'dataset_landing_url': ''}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {},
   "outputs": [],
   "source": [
    "infile = '/Users/denis/Data/archs-v1.1/series/GSE24565-GPL11154.txt'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gse</th>\n",
       "      <th>variable</th>\n",
       "      <th>value</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>GSM897081</td>\n",
       "      <td>biomaterial_type</td>\n",
       "      <td>immortalized cell line</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>GSM973667</td>\n",
       "      <td>biomaterial_type</td>\n",
       "      <td>immortalized cell line</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>GSM973675</td>\n",
       "      <td>biomaterial_type</td>\n",
       "      <td>immortalized cell line</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>GSM973676</td>\n",
       "      <td>biomaterial_type</td>\n",
       "      <td>immortalized cell line</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>GSM973677</td>\n",
       "      <td>biomaterial_type</td>\n",
       "      <td>immortalized cell line</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>GSM973690</td>\n",
       "      <td>biomaterial_type</td>\n",
       "      <td>immortalized cell line</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41</th>\n",
       "      <td>GSM977034</td>\n",
       "      <td>biomaterial_type</td>\n",
       "      <td>primary cell</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43</th>\n",
       "      <td>GSM977036</td>\n",
       "      <td>biomaterial_type</td>\n",
       "      <td>primary cell</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>55</th>\n",
       "      <td>GSM977050</td>\n",
       "      <td>biomaterial_type</td>\n",
       "      <td>primary cell</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>58</th>\n",
       "      <td>GSM897081</td>\n",
       "      <td>biorep</td>\n",
       "      <td>015WC,016WC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>59</th>\n",
       "      <td>GSM897083</td>\n",
       "      <td>biorep</td>\n",
       "      <td>021WC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>60</th>\n",
       "      <td>GSM973661</td>\n",
       "      <td>biorep</td>\n",
       "      <td>087N,088N</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>61</th>\n",
       "      <td>GSM973662</td>\n",
       "      <td>biorep</td>\n",
       "      <td>043WC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>62</th>\n",
       "      <td>GSM973663</td>\n",
       "      <td>biorep</td>\n",
       "      <td>085N,086N</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>63</th>\n",
       "      <td>GSM973664</td>\n",
       "      <td>biorep</td>\n",
       "      <td>087C,088C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>64</th>\n",
       "      <td>GSM973665</td>\n",
       "      <td>biorep</td>\n",
       "      <td>085N,086N</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>65</th>\n",
       "      <td>GSM973666</td>\n",
       "      <td>biorep</td>\n",
       "      <td>085C,086C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>66</th>\n",
       "      <td>GSM973667</td>\n",
       "      <td>biorep</td>\n",
       "      <td>085WC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>67</th>\n",
       "      <td>GSM973668</td>\n",
       "      <td>biorep</td>\n",
       "      <td>087WC,088WC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>68</th>\n",
       "      <td>GSM973669</td>\n",
       "      <td>biorep</td>\n",
       "      <td>087C,088C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>69</th>\n",
       "      <td>GSM973670</td>\n",
       "      <td>biorep</td>\n",
       "      <td>045C,046C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>70</th>\n",
       "      <td>GSM973671</td>\n",
       "      <td>biorep</td>\n",
       "      <td>045C,046C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>71</th>\n",
       "      <td>GSM973672</td>\n",
       "      <td>biorep</td>\n",
       "      <td>089C,090C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>72</th>\n",
       "      <td>GSM973673</td>\n",
       "      <td>biorep</td>\n",
       "      <td>045N,046N</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>73</th>\n",
       "      <td>GSM973674</td>\n",
       "      <td>biorep</td>\n",
       "      <td>087WC,088WC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>74</th>\n",
       "      <td>GSM973675</td>\n",
       "      <td>biorep</td>\n",
       "      <td>085WC,086WC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>75</th>\n",
       "      <td>GSM973676</td>\n",
       "      <td>biorep</td>\n",
       "      <td>089WC,090WC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>76</th>\n",
       "      <td>GSM973677</td>\n",
       "      <td>biorep</td>\n",
       "      <td>085WC,086WC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>77</th>\n",
       "      <td>GSM973678</td>\n",
       "      <td>biorep</td>\n",
       "      <td>045N,046N</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>78</th>\n",
       "      <td>GSM973679</td>\n",
       "      <td>biorep</td>\n",
       "      <td>045N</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1072</th>\n",
       "      <td>GSM973687</td>\n",
       "      <td>rnaextract description</td>\n",
       "      <td>Rna shorter than 200 nt that has not been sepe...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1073</th>\n",
       "      <td>GSM973688</td>\n",
       "      <td>rnaextract description</td>\n",
       "      <td>Rna shorter than 200 nt that has not been sepe...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1074</th>\n",
       "      <td>GSM973689</td>\n",
       "      <td>rnaextract description</td>\n",
       "      <td>Rna shorter than 200 nt that has not been sepe...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1075</th>\n",
       "      <td>GSM973690</td>\n",
       "      <td>rnaextract description</td>\n",
       "      <td>Rna shorter than 200 nt that has not been sepe...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1076</th>\n",
       "      <td>GSM973691</td>\n",
       "      <td>rnaextract description</td>\n",
       "      <td>Rna shorter than 200 nt that has not been sepe...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1077</th>\n",
       "      <td>GSM973692</td>\n",
       "      <td>rnaextract description</td>\n",
       "      <td>Rna shorter than 200 nt that has not been sepe...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1078</th>\n",
       "      <td>GSM973693</td>\n",
       "      <td>rnaextract description</td>\n",
       "      <td>Rna shorter than 200 nt that has not been sepe...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1079</th>\n",
       "      <td>GSM977028</td>\n",
       "      <td>rnaextract description</td>\n",
       "      <td>Rna shorter than 200 nt that has not been sepe...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1080</th>\n",
       "      <td>GSM977029</td>\n",
       "      <td>rnaextract description</td>\n",
       "      <td>Rna shorter than 200 nt that has not been sepe...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1081</th>\n",
       "      <td>GSM977030</td>\n",
       "      <td>rnaextract description</td>\n",
       "      <td>Rna shorter than 200 nt that has not been sepe...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1082</th>\n",
       "      <td>GSM977031</td>\n",
       "      <td>rnaextract description</td>\n",
       "      <td>Rna shorter than 200 nt that has not been sepe...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1083</th>\n",
       "      <td>GSM977032</td>\n",
       "      <td>rnaextract description</td>\n",
       "      <td>Rna shorter than 200 nt that has not been sepe...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1084</th>\n",
       "      <td>GSM977033</td>\n",
       "      <td>rnaextract description</td>\n",
       "      <td>Rna shorter than 200 nt that has not been sepe...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1085</th>\n",
       "      <td>GSM977034</td>\n",
       "      <td>rnaextract description</td>\n",
       "      <td>Rna shorter than 200 nt that has not been sepe...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1086</th>\n",
       "      <td>GSM977035</td>\n",
       "      <td>rnaextract description</td>\n",
       "      <td>Rna shorter than 200 nt that has not been sepe...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1087</th>\n",
       "      <td>GSM977036</td>\n",
       "      <td>rnaextract description</td>\n",
       "      <td>Rna shorter than 200 nt that has not been sepe...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1088</th>\n",
       "      <td>GSM977037</td>\n",
       "      <td>rnaextract description</td>\n",
       "      <td>Rna shorter than 200 nt that has not been sepe...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1089</th>\n",
       "      <td>GSM977038</td>\n",
       "      <td>rnaextract description</td>\n",
       "      <td>Rna shorter than 200 nt that has not been sepe...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1090</th>\n",
       "      <td>GSM977039</td>\n",
       "      <td>rnaextract description</td>\n",
       "      <td>Rna shorter than 200 nt that has not been sepe...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1091</th>\n",
       "      <td>GSM977040</td>\n",
       "      <td>rnaextract description</td>\n",
       "      <td>Rna shorter than 200 nt that has not been sepe...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1092</th>\n",
       "      <td>GSM977041</td>\n",
       "      <td>rnaextract description</td>\n",
       "      <td>Rna shorter than 200 nt that has not been sepe...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1093</th>\n",
       "      <td>GSM977042</td>\n",
       "      <td>rnaextract description</td>\n",
       "      <td>Rna shorter than 200 nt that has not been sepe...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1094</th>\n",
       "      <td>GSM977045</td>\n",
       "      <td>rnaextract description</td>\n",
       "      <td>Rna shorter than 200 nt that has not been sepe...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1095</th>\n",
       "      <td>GSM977046</td>\n",
       "      <td>rnaextract description</td>\n",
       "      <td>Rna shorter than 200 nt that has not been sepe...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1096</th>\n",
       "      <td>GSM977047</td>\n",
       "      <td>rnaextract description</td>\n",
       "      <td>Rna shorter than 200 nt that has not been sepe...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1097</th>\n",
       "      <td>GSM977048</td>\n",
       "      <td>rnaextract description</td>\n",
       "      <td>Rna shorter than 200 nt that has not been sepe...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1098</th>\n",
       "      <td>GSM977049</td>\n",
       "      <td>rnaextract description</td>\n",
       "      <td>Rna shorter than 200 nt that has not been sepe...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1099</th>\n",
       "      <td>GSM977050</td>\n",
       "      <td>rnaextract description</td>\n",
       "      <td>Rna shorter than 200 nt that has not been sepe...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1100</th>\n",
       "      <td>GSM977051</td>\n",
       "      <td>rnaextract description</td>\n",
       "      <td>Rna shorter than 200 nt that has not been sepe...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1101</th>\n",
       "      <td>GSM977052</td>\n",
       "      <td>rnaextract description</td>\n",
       "      <td>Rna shorter than 200 nt that has not been sepe...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>830 rows × 3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "            gse                variable  \\\n",
       "0     GSM897081        biomaterial_type   \n",
       "8     GSM973667        biomaterial_type   \n",
       "16    GSM973675        biomaterial_type   \n",
       "17    GSM973676        biomaterial_type   \n",
       "18    GSM973677        biomaterial_type   \n",
       "31    GSM973690        biomaterial_type   \n",
       "41    GSM977034        biomaterial_type   \n",
       "43    GSM977036        biomaterial_type   \n",
       "55    GSM977050        biomaterial_type   \n",
       "58    GSM897081                  biorep   \n",
       "59    GSM897083                  biorep   \n",
       "60    GSM973661                  biorep   \n",
       "61    GSM973662                  biorep   \n",
       "62    GSM973663                  biorep   \n",
       "63    GSM973664                  biorep   \n",
       "64    GSM973665                  biorep   \n",
       "65    GSM973666                  biorep   \n",
       "66    GSM973667                  biorep   \n",
       "67    GSM973668                  biorep   \n",
       "68    GSM973669                  biorep   \n",
       "69    GSM973670                  biorep   \n",
       "70    GSM973671                  biorep   \n",
       "71    GSM973672                  biorep   \n",
       "72    GSM973673                  biorep   \n",
       "73    GSM973674                  biorep   \n",
       "74    GSM973675                  biorep   \n",
       "75    GSM973676                  biorep   \n",
       "76    GSM973677                  biorep   \n",
       "77    GSM973678                  biorep   \n",
       "78    GSM973679                  biorep   \n",
       "...         ...                     ...   \n",
       "1072  GSM973687  rnaextract description   \n",
       "1073  GSM973688  rnaextract description   \n",
       "1074  GSM973689  rnaextract description   \n",
       "1075  GSM973690  rnaextract description   \n",
       "1076  GSM973691  rnaextract description   \n",
       "1077  GSM973692  rnaextract description   \n",
       "1078  GSM973693  rnaextract description   \n",
       "1079  GSM977028  rnaextract description   \n",
       "1080  GSM977029  rnaextract description   \n",
       "1081  GSM977030  rnaextract description   \n",
       "1082  GSM977031  rnaextract description   \n",
       "1083  GSM977032  rnaextract description   \n",
       "1084  GSM977033  rnaextract description   \n",
       "1085  GSM977034  rnaextract description   \n",
       "1086  GSM977035  rnaextract description   \n",
       "1087  GSM977036  rnaextract description   \n",
       "1088  GSM977037  rnaextract description   \n",
       "1089  GSM977038  rnaextract description   \n",
       "1090  GSM977039  rnaextract description   \n",
       "1091  GSM977040  rnaextract description   \n",
       "1092  GSM977041  rnaextract description   \n",
       "1093  GSM977042  rnaextract description   \n",
       "1094  GSM977045  rnaextract description   \n",
       "1095  GSM977046  rnaextract description   \n",
       "1096  GSM977047  rnaextract description   \n",
       "1097  GSM977048  rnaextract description   \n",
       "1098  GSM977049  rnaextract description   \n",
       "1099  GSM977050  rnaextract description   \n",
       "1100  GSM977051  rnaextract description   \n",
       "1101  GSM977052  rnaextract description   \n",
       "\n",
       "                                                  value  \n",
       "0                                immortalized cell line  \n",
       "8                                immortalized cell line  \n",
       "16                               immortalized cell line  \n",
       "17                               immortalized cell line  \n",
       "18                               immortalized cell line  \n",
       "31                               immortalized cell line  \n",
       "41                                         primary cell  \n",
       "43                                         primary cell  \n",
       "55                                         primary cell  \n",
       "58                                          015WC,016WC  \n",
       "59                                                021WC  \n",
       "60                                            087N,088N  \n",
       "61                                                043WC  \n",
       "62                                            085N,086N  \n",
       "63                                            087C,088C  \n",
       "64                                            085N,086N  \n",
       "65                                            085C,086C  \n",
       "66                                                085WC  \n",
       "67                                          087WC,088WC  \n",
       "68                                            087C,088C  \n",
       "69                                            045C,046C  \n",
       "70                                            045C,046C  \n",
       "71                                            089C,090C  \n",
       "72                                            045N,046N  \n",
       "73                                          087WC,088WC  \n",
       "74                                          085WC,086WC  \n",
       "75                                          089WC,090WC  \n",
       "76                                          085WC,086WC  \n",
       "77                                            045N,046N  \n",
       "78                                                 045N  \n",
       "...                                                 ...  \n",
       "1072  Rna shorter than 200 nt that has not been sepe...  \n",
       "1073  Rna shorter than 200 nt that has not been sepe...  \n",
       "1074  Rna shorter than 200 nt that has not been sepe...  \n",
       "1075  Rna shorter than 200 nt that has not been sepe...  \n",
       "1076  Rna shorter than 200 nt that has not been sepe...  \n",
       "1077  Rna shorter than 200 nt that has not been sepe...  \n",
       "1078  Rna shorter than 200 nt that has not been sepe...  \n",
       "1079  Rna shorter than 200 nt that has not been sepe...  \n",
       "1080  Rna shorter than 200 nt that has not been sepe...  \n",
       "1081  Rna shorter than 200 nt that has not been sepe...  \n",
       "1082  Rna shorter than 200 nt that has not been sepe...  \n",
       "1083  Rna shorter than 200 nt that has not been sepe...  \n",
       "1084  Rna shorter than 200 nt that has not been sepe...  \n",
       "1085  Rna shorter than 200 nt that has not been sepe...  \n",
       "1086  Rna shorter than 200 nt that has not been sepe...  \n",
       "1087  Rna shorter than 200 nt that has not been sepe...  \n",
       "1088  Rna shorter than 200 nt that has not been sepe...  \n",
       "1089  Rna shorter than 200 nt that has not been sepe...  \n",
       "1090  Rna shorter than 200 nt that has not been sepe...  \n",
       "1091  Rna shorter than 200 nt that has not been sepe...  \n",
       "1092  Rna shorter than 200 nt that has not been sepe...  \n",
       "1093  Rna shorter than 200 nt that has not been sepe...  \n",
       "1094  Rna shorter than 200 nt that has not been sepe...  \n",
       "1095  Rna shorter than 200 nt that has not been sepe...  \n",
       "1096  Rna shorter than 200 nt that has not been sepe...  \n",
       "1097  Rna shorter than 200 nt that has not been sepe...  \n",
       "1098  Rna shorter than 200 nt that has not been sepe...  \n",
       "1099  Rna shorter than 200 nt that has not been sepe...  \n",
       "1100  Rna shorter than 200 nt that has not been sepe...  \n",
       "1101  Rna shorter than 200 nt that has not been sepe...  \n",
       "\n",
       "[830 rows x 3 columns]"
      ]
     },
     "execution_count": 43,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sample_metadata_dataframe = pd.concat([pd.melt(pd.read_table(x).rename(columns={'Unnamed: 0': 'gsm'}), id_vars='gsm').dropna() for x in infiles])\n",
    "sample_metadata"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "hide_input": false,
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
