{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import os, json\n",
    "import pymysql\n",
    "pymysql.install_as_MySQLdb()\n",
    "import pandas as pd\n",
    "from sqlalchemy import create_engine\n",
    "\n",
    "engine = create_engine(os.environ['SQLALCHEMY_DATABASE_URI'])\n",
    "\n",
    "data = {x: pd.read_sql_table(x, engine) for x in ['tool', 'parameter', 'parameter_value']}\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>id</th>\n",
       "      <th>tool_string</th>\n",
       "      <th>tool_name</th>\n",
       "      <th>tool_description</th>\n",
       "      <th>introduction</th>\n",
       "      <th>methods</th>\n",
       "      <th>reference</th>\n",
       "      <th>reference_link</th>\n",
       "      <th>default_selected</th>\n",
       "      <th>section_fk</th>\n",
       "      <th>input</th>\n",
       "      <th>display</th>\n",
       "      <th>result_description</th>\n",
       "      <th>video_tutorial</th>\n",
       "      <th>requires_signature</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>pca</td>\n",
       "      <td>PCA</td>\n",
       "      <td>Linear dimensionality reduction technique to v...</td>\n",
       "      <td>Principal Component Analysis (PCA) is a statis...</td>\n",
       "      <td>Principal Component Analysis was performed usi...</td>\n",
       "      <td>Pearson, K. (1901). &lt;b&gt;LIII. On lines and plan...</td>\n",
       "      <td>https://doi.org/10.1080/14786440109462720</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>dataset</td>\n",
       "      <td>1</td>\n",
       "      <td>The PCA plug-in embeds an interactive, three-d...</td>\n",
       "      <td>https://www.youtube.com/watch?v=r7WYtxeAaDw</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>2</td>\n",
       "      <td>tsne</td>\n",
       "      <td>t-SNE</td>\n",
       "      <td>Nonlinear dimensionality reduction technique t...</td>\n",
       "      <td>t-distributed Stochastic Neighbor Embedding (t...</td>\n",
       "      <td>t-SNE analysis was performed using the `TSNE` ...</td>\n",
       "      <td>L.J.P. van der Maaten and G.E. Hinton. &lt;b&gt;Visu...</td>\n",
       "      <td>https://lvdmaaten.github.io/publications/paper...</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>dataset</td>\n",
       "      <td>0</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>3</td>\n",
       "      <td>clustergrammer</td>\n",
       "      <td>Clustergrammer</td>\n",
       "      <td>Interactive hierarchical clustering heatmap vi...</td>\n",
       "      <td>Clustergrammer is a web-based tool for visuali...</td>\n",
       "      <td>The interactive heatmap was generated using Cl...</td>\n",
       "      <td>Fernandez, N.F., Gundersen, G.W., Rahman, A., ...</td>\n",
       "      <td>http://dx.doi.org/10.1038/sdata.2017.151</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>dataset</td>\n",
       "      <td>1</td>\n",
       "      <td>The Clustergrammer plug-in embeds an interacti...</td>\n",
       "      <td>https://youtu.be/8MgvyeroW98?t=59s</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>4</td>\n",
       "      <td>library_size_analysis</td>\n",
       "      <td>Library Size Analysis</td>\n",
       "      <td>Analysis of readcount distribution for the sam...</td>\n",
       "      <td>In order to quantify gene expression in an RNA...</td>\n",
       "      <td>Read counts were calculated by performing the ...</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>dataset</td>\n",
       "      <td>1</td>\n",
       "      <td>The Library Size Analysis plug-in embeds an in...</td>\n",
       "      <td>None</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>5</td>\n",
       "      <td>signature_table</td>\n",
       "      <td>Differential Expression Table</td>\n",
       "      <td>Differential expression analysis between two g...</td>\n",
       "      <td>Gene expression signatures are alterations in ...</td>\n",
       "      <td>The gene expression signature was generated by...</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>signature</td>\n",
       "      <td>1</td>\n",
       "      <td>The Signature Table plug-in embeds an interact...</td>\n",
       "      <td>None</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   id            tool_string                      tool_name  \\\n",
       "0   1                    pca                            PCA   \n",
       "1   2                   tsne                          t-SNE   \n",
       "2   3         clustergrammer                 Clustergrammer   \n",
       "3   4  library_size_analysis          Library Size Analysis   \n",
       "4   5        signature_table  Differential Expression Table   \n",
       "\n",
       "                                    tool_description  \\\n",
       "0  Linear dimensionality reduction technique to v...   \n",
       "1  Nonlinear dimensionality reduction technique t...   \n",
       "2  Interactive hierarchical clustering heatmap vi...   \n",
       "3  Analysis of readcount distribution for the sam...   \n",
       "4  Differential expression analysis between two g...   \n",
       "\n",
       "                                        introduction  \\\n",
       "0  Principal Component Analysis (PCA) is a statis...   \n",
       "1  t-distributed Stochastic Neighbor Embedding (t...   \n",
       "2  Clustergrammer is a web-based tool for visuali...   \n",
       "3  In order to quantify gene expression in an RNA...   \n",
       "4  Gene expression signatures are alterations in ...   \n",
       "\n",
       "                                             methods  \\\n",
       "0  Principal Component Analysis was performed usi...   \n",
       "1  t-SNE analysis was performed using the `TSNE` ...   \n",
       "2  The interactive heatmap was generated using Cl...   \n",
       "3  Read counts were calculated by performing the ...   \n",
       "4  The gene expression signature was generated by...   \n",
       "\n",
       "                                           reference  \\\n",
       "0  Pearson, K. (1901). <b>LIII. On lines and plan...   \n",
       "1  L.J.P. van der Maaten and G.E. Hinton. <b>Visu...   \n",
       "2  Fernandez, N.F., Gundersen, G.W., Rahman, A., ...   \n",
       "3                                                      \n",
       "4                                                      \n",
       "\n",
       "                                      reference_link  default_selected  \\\n",
       "0          https://doi.org/10.1080/14786440109462720                 1   \n",
       "1  https://lvdmaaten.github.io/publications/paper...                 0   \n",
       "2           http://dx.doi.org/10.1038/sdata.2017.151                 1   \n",
       "3                                                                    1   \n",
       "4                                                                    0   \n",
       "\n",
       "   section_fk      input  display  \\\n",
       "0           1    dataset        1   \n",
       "1           1    dataset        0   \n",
       "2           1    dataset        1   \n",
       "3           1    dataset        1   \n",
       "4           2  signature        1   \n",
       "\n",
       "                                  result_description  \\\n",
       "0  The PCA plug-in embeds an interactive, three-d...   \n",
       "1                                                      \n",
       "2  The Clustergrammer plug-in embeds an interacti...   \n",
       "3  The Library Size Analysis plug-in embeds an in...   \n",
       "4  The Signature Table plug-in embeds an interact...   \n",
       "\n",
       "                                video_tutorial  requires_signature  \n",
       "0  https://www.youtube.com/watch?v=r7WYtxeAaDw                   0  \n",
       "1                                         None                   0  \n",
       "2           https://youtu.be/8MgvyeroW98?t=59s                   0  \n",
       "3                                         None                   0  \n",
       "4                                         None                   1  "
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data['tool'].head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>id</th>\n",
       "      <th>parameter_name</th>\n",
       "      <th>parameter_description</th>\n",
       "      <th>parameter_string</th>\n",
       "      <th>tool_fk</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>2</td>\n",
       "      <td>Top Genes</td>\n",
       "      <td>Number of most variably expressed genes to use...</td>\n",
       "      <td>nr_genes</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>3</td>\n",
       "      <td>Normalization</td>\n",
       "      <td>Normalization method for the dataset</td>\n",
       "      <td>normalization</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>5</td>\n",
       "      <td>Top Genes</td>\n",
       "      <td>Number of most variably expressed genes to use...</td>\n",
       "      <td>nr_genes</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>6</td>\n",
       "      <td>Normalization</td>\n",
       "      <td>Normalization method for the dataset</td>\n",
       "      <td>normalization</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>7</td>\n",
       "      <td>Top Genes</td>\n",
       "      <td>Number of most variably expressed genes to use...</td>\n",
       "      <td>nr_genes</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   id parameter_name                              parameter_description  \\\n",
       "0   2      Top Genes  Number of most variably expressed genes to use...   \n",
       "1   3  Normalization               Normalization method for the dataset   \n",
       "2   5      Top Genes  Number of most variably expressed genes to use...   \n",
       "3   6  Normalization               Normalization method for the dataset   \n",
       "4   7      Top Genes  Number of most variably expressed genes to use...   \n",
       "\n",
       "  parameter_string  tool_fk  \n",
       "0         nr_genes        1  \n",
       "1    normalization        1  \n",
       "2         nr_genes        2  \n",
       "3    normalization        2  \n",
       "4         nr_genes        3  "
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data['parameter'].head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>id</th>\n",
       "      <th>parameter_fk</th>\n",
       "      <th>value</th>\n",
       "      <th>type</th>\n",
       "      <th>default</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>3</td>\n",
       "      <td>2</td>\n",
       "      <td>500</td>\n",
       "      <td>int</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>4</td>\n",
       "      <td>2</td>\n",
       "      <td>2500</td>\n",
       "      <td>int</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>5</td>\n",
       "      <td>2</td>\n",
       "      <td>5000</td>\n",
       "      <td>int</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>6</td>\n",
       "      <td>3</td>\n",
       "      <td>logCPM</td>\n",
       "      <td>str</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>7</td>\n",
       "      <td>3</td>\n",
       "      <td>quantile</td>\n",
       "      <td>str</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   id  parameter_fk     value type  default\n",
       "0   3             2       500  int        0\n",
       "1   4             2      2500  int        1\n",
       "2   5             2      5000  int        0\n",
       "3   6             3    logCPM  str        1\n",
       "4   7             3  quantile  str        0"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data['parameter_value'].head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "hide_input": false,
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
